nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenHTTP.java

```diff
@@ -351,6 +372,60 @@ synchronized private void createHttpServerFromService(final ProcessContext conte
         initialized.set(true);
     }
 
+    private ServerConnector createServerConnector(Server server, int port, SSLContextService sslContextService, boolean sslRequired, boolean needClientAuth) {
+        final SslContextFactory contextFactory = createSslContextFactory(sslContextService, sslRequired, needClientAuth);
+        return createServerConnector(server, port, contextFactory, sslRequired);
+    }
+
+    private ServerConnector createServerConnector(Server server, int port, SslContextFactory contextFactory, boolean sslRequired) {
+        final ServerConnector connector;
+        final HttpConfiguration httpConfiguration = new HttpConfiguration();
+        if (!sslRequired) {
```

Comment:
Changing the logic to check if (sslRequired) { ... } else { ... } would be more straightforward.